xblock/fields.py

```diff
@@ -824,9 +833,23 @@ class String(JSONField):
     """
     MUTABLE = False
 
+    def _sanitize(self, value):
+        """
+        Remove the control characters that are not allowed in XML:
+        https://www.w3.org/TR/xml/#charsets
+        Leave all other characters.
+        """
+        if isinstance(value, unicode):
+            new_value = u''.join(ch for ch in value if unicodedata.category(ch)[0] != u'C' or ch in (u'\n', u'\r', u'\t'))
+        elif isinstance(value, str):
+            new_value = ''.join(ch for ch in value if ord(ch) >= 32 or ch in ('\n', '\r', '\t'))
+        else:
+            return value
+        return value if value == new_value else new_value
```

Comment:
Mutability, in this case, is about whether a pattern like this is possible:

    value = MutableObject()
    xblock.field = value
    value.mutate()
    xblock.field.read_mutation() # what does this show?

For mutable objects (like lists and dicts), it's important to preserve object identity so that mutations that happen outside the field context are still reflected in the field.

For strings, however, the objects themselves are immutable. You can get different strings by calling methods (like `replace`), but those won't have the same object identity.